pkg/identity/identitysync/one-login.go

```diff
@@ -0,0 +1,333 @@
+	req, _ := http.NewRequest("POST", url, bytes.NewBuffer(jsonStr))
+	req.Header.Add("Authorization", "client_id: "+s.clientID.Get()+",client_secret: "+s.clientSecret.Get())
+	req.Header.Add("content-type", "application/json")
+
+	res, err := http.DefaultClient.Do(req)
+	if err != nil {
+		return err
+	}
+	b, err := io.ReadAll(res.Body)
+	if err != nil {
+		return err
+	}
+
+	//return the error if its anything but a 200
+	if res.StatusCode != 200 {
+		return fmt.Errorf(string(b))
+	}
+
+	var lu GetAccessTokenResponse
+	err = json.Unmarshal(b, &lu)
+	if err != nil {
+		return err
+	}
+
+	s.token.Set(lu.AccessToken)
+
+	return nil
+}
+func (s *OneLoginSync) TestConfig(ctx context.Context) error {
+	_, err := s.ListUsers(ctx)
+	if err != nil {
+		return errors.Wrap(err, "failed to list users while testing one login identity provider configuration")
+	}
+	_, err = s.ListGroups(ctx)
+	if err != nil {
+		return errors.Wrap(err, "failed to list groups while testing one login identity provider configuration")
+	}
+	return nil
+}
+
+// userFromOktaUser converts a Okta user to the identityprovider interface user type
+func (o *OneLoginSync) idpUserFromOneLoginUser(ctx context.Context, oneLoginUser *OneLoginUser) (identity.IDPUser, error) {
+	u := identity.IDPUser{
+		ID:        strconv.Itoa(oneLoginUser.ID),
+		FirstName: oneLoginUser.Firstname,
+		LastName:  oneLoginUser.Lastname,
+		Email:     oneLoginUser.Email,
+		Groups:    []string{},
+	}
+
+	for _, r := range oneLoginUser.RoleID {
+		u.Groups = append(u.Groups, strconv.Itoa(r))
+	}
+
+	return u, nil
+}
+
+func (o *OneLoginSync) idpGroupFromOneLoginGroup(oneLoginGroup OneLoginGroup) identity.IDPGroup {
+	return identity.IDPGroup{
+		ID:   strconv.Itoa(oneLoginGroup.ID),
+		Name: oneLoginGroup.Name,
+	}
+
+}
+
+func (a *OneLoginSync) ListUsers(ctx context.Context) ([]identity.IDPUser, error) {
+
+	//get all users
+	idpUsers := []identity.IDPUser{}
+	hasMore := true
+	var nextToken *string
+	url := OneLoginBaseURL + "/api/1/users"
+	for hasMore {
+
+		req, _ := http.NewRequest("GET", url, nil)
+		req.Header.Add("Authorization", "Bearer: "+a.token.Get())
+
+		res, err := http.DefaultClient.Do(req)
+		if err != nil {
+			return nil, err
+		}
+		b, err := io.ReadAll(res.Body)
+		if err != nil {
+			return nil, err
+		}
+
+		//return the error if its anything but a 200
+		if res.StatusCode != 200 {
+			return nil, fmt.Errorf(string(b))
+		}
+
+		var lu OneLoginListUserResponse
+		err = json.Unmarshal(b, &lu)
+		if err != nil {
+			return nil, err
+		}
+		for _, u := range lu.Users {
+
+			user, err := a.idpUserFromOneLoginUser(ctx, &u)
+			if err != nil {
+				return nil, err
+			}
+			idpUsers = append(idpUsers, user)
+		}
+		nextToken = lu.Pagination.NextLink
+		if nextToken != nil {
+			url = *nextToken
+		} else {
+			hasMore = false
+		}
+
+	}
+	return idpUsers, nil
+}
+
+func (a *OneLoginSync) ListGroups(ctx context.Context) ([]identity.IDPGroup, error) {
+	idpGroups := []identity.IDPGroup{}
```

Comment:
nitpick: initialising variables to zero value in go
```suggestion
	var idpGroups []identity.IDPGroup
```